### ISSUE TITLE
Query about pharmacophore generation

### ISSUE THREAD

**Issue by dww100** [tmcgrath325/MolecularGaussians.jl#11]
Hello,

This is a really nice tool, thanks for the hard work.

I am looking to use the tool with molecules output from generative models which do not come with pharmacophore information in the "PubChem" format. Do you know how this could be calculated? I see the OEChem label in the header but do not have access to the OpenEye toolkit - so information on the format would be great.

Looking at the example data it looks like it it something like:
```
> <PUBCHEM_PHARMACOPHORE_FEATURES>
A
X Y acceptor
```

Where
`A` = number of pharmacophore features
`X` = number of atoms in feature
`Y` = atom index(es) in feature
with `acceptor` being the feature type. Is that right?

Is there a list of features somewhere?

Cheers
Dave

**Comment by tmcgrath325** [comment]
> Looking at the example data it looks like it it something like:
>
> > <PUBCHEM_PHARMACOPHORE_FEATURES>
> A
> X Y acceptor
> 
> Where
> A = number of pharmacophore features
> X = number of atoms in feature
> Y = atom index(es) in feature
> with acceptor being the feature type. Is that right?

That's correct. As of yet, I've only implemented a way to read pharmacophore features from an SDF file in the PubChem format, which uses the [features used in PUG REST](https://pubchemdocs.ncbi.nlm.nih.gov/pug-rest#_Toc494865567) (i.e. acceptors, donors, rings, cations, anions, and hydrophobes). 

I don't have an implementation for this yet, but I think it should be relatively straightforward to use [SMARTS](https://daylight.com/dayhtml_tutorials/languages/smarts/smarts_examples.html) strings with [MolecularGraph.jl ](https://github.com/mojaie/MolecularGraph.jl) to identify features of interest. These wouldn't necessarily have to be the same kind of features as the PubChem ones. 

 With the way things are set up currently, you would have to store the features of interest as a vector of strings in a molecule's  `attributes` with the key `PUBCHEM_PHARMACOPHORE_FEATURES` with the above format. Then, the features could be accessed by the `pharmfeatures` method and used to build a `PharmacphoreGMM`. This sounds somewhat painful to do, though, and I'll need to make some changes to allow for more flexibility.

I've been working on some improvements to the alignment optimization, and I hope to revisit this package and clean up the API once I've finished up with that.

-Tom

**Comment by dww100** [comment]
Thanks for the update - look forwards to the developments
